cognite/client/_api/synthetic_time_series.py

```diff
@@ -46,40 +45,57 @@ def retrieve(
+                >>> dps = c.datapoints.synthetic.retrieve(expressions=["TS{id:123} + TS{externalId:'abc'}"], start="2w-ago", end="now")
 
             Use variables to re-use an expression:
 
-                >>> from cognite.client.experimental import CogniteClient
+                >>> from cognite.client import CogniteClient
                 >>> c = CogniteClient()
                 >>> vars = {"A": "my_ts_external_id", "B": client.time_series.retrieve(id=1)}
-                >>> dps = c.datapoints.synthetic.retrieve(expression="A+B", start="2w-ago", end="now", variables=vars)
+                >>> dps = c.datapoints.synthetic.retrieve(expressions=["A+B"], start="2w-ago", end="now", variables=vars)
 
             Use sympy to build complex expressions:
 
-                >>> from cognite.client.experimental import CogniteClient
+                >>> from cognite.client import CogniteClient
                 >>> c = CogniteClient()
                 >>> from sympy import symbols, cos, pi
                 >>> a = sympy.symbols('a')
-                >>> dps = c.datapoints.synthetic.retrieve(pi * cos(a), start="2w-ago", end="now", variables={"a": "my_ts_external_id"},aggregate='interpolation',granularity='1m')
+                >>> dps = c.datapoints.synthetic.retrieve([pi * cos(a)], start="2w-ago", end="now", variables={"a": "my_ts_external_id"},aggregate='interpolation',granularity='1m')
             """
         if limit is None or limit == -1:
             limit = float("inf")
-        expression, short_expression = SyntheticDatapointsAPI._build_expression(
-            expression, variables, aggregate, granularity
+
+        tasks = []
+
+        for i in range(len(expressions)):
+            expression, short_expression = SyntheticDatapointsAPI._build_expression(
+                expressions[i], variables, aggregate, granularity
+            )
+            query = {
+                "expression": expression,
+                "start": cognite.client.utils._time.timestamp_to_ms(start),
+                "end": cognite.client.utils._time.timestamp_to_ms(end),
+            }
+            query_datapoints = Datapoints(value=[], error=[])
+            query_datapoints.external_id = short_expression
+
+            tasks.append((query, query_datapoints, limit))
+
+        datapoints_summary = utils._concurrency.execute_tasks_concurrently(
+            self.func, tasks, max_workers=self._config.max_workers
         )
-        query = {
-            "expression": expression,
-            "start": cognite.client.utils._time.timestamp_to_ms(start),
-            "end": cognite.client.utils._time.timestamp_to_ms(end),
-        }
-        datapoints = Datapoints(value=[], error=[])
-        datapoints.external_id = short_expression  # for dataframe readability
+
+        if datapoints_summary.exceptions:
+            raise datapoints_summary.exceptions[0]
+
+        return datapoints_summary.results
+
+    def func(self, query, datapoints, limit):
```

Comment:
func? please make this a sensibly named method starting with _

Reply:
ah, missed to change it, thx